apps/s_client.c

```diff
@@ -3481,8 +3481,9 @@ static void print_stuff(BIO *bio, SSL *s, int full)
     print_verify_detail(s, bio);
     BIO_printf(bio, (SSL_session_reused(s) ? "---\nReused, " : "---\nNew, "));
     c = SSL_get_current_cipher(s);
-    BIO_printf(bio, "%s, Cipher is %s\n",
-               SSL_CIPHER_get_version(c), SSL_CIPHER_get_name(c));
+    BIO_printf(bio, "Cipher is %s (%s)\n",
+               SSL_CIPHER_get_name(c), SSL_CIPHER_get_version(c));
```

Comment:
Maybe for now just keep the Cipher output as is?

Reply:
Yes, I can do that.

I'm tempted to solve it like this: 😉
```c
    if (OSSL_PROVIDER_available(NULL, "gostprov"))
        BIO_printf(bio, "%s, Cipher is %s\n",
                   SSL_CIPHER_get_version(c), SSL_CIPHER_get_name(c));
    else
        BIO_printf(bio, "Cipher is %s (%s)\n",
                   SSL_CIPHER_get_name(c), SSL_CIPHER_get_version(c));
```

Reply:
No, that would not be a solution - that would be an unacceptable hack.